flocker_bb/builders/flocker.py

```diff
@@ -328,9 +335,28 @@ def makeInternalDocsFactory():
             'docs',
             ],
         path="private_html",
-        haltOnFailure=True,
         doStepIf=isMasterBranch('flocker'),
         ))
+    factory.addStep(MasterShellCommand(
+        name='upload-release-documentation',
+        description=["uploading", "release", "documentation"],
+        descriptionDone=["upload", "release", "documentation"],
+        command=[
+            # We use s3cmd instead of gsutil here because of
+            # https://github.com/GoogleCloudPlatform/gsutil/issues/247
+            "s3cmd", "sync",
+            '--verbose',
+            '--delete-removed',
+            '--no-preserve',
+            Interpolate('%s/%s/docs/' % (branch, revision)),
+            Interpolate(
+                "s3://%(kw:bucket)s/%(prop:version)s/",
+                bucket='clusterhq-dev-docs',
+            ),
+        ],
+        path="private_html",
+        doStepIf=isReleaseBranch('flocker'),
```

Comment:
We could just use this upload as the place for built docs, but since this was in the critical pass for other things, I haven't bothered to do that here.